packages/story-editor/src/components/mediaRecording/provider.js

```diff
@@ -134,8 +139,8 @@ function MediaRecordingProvider({ children }) {
     // granted permission, so it's not easy to detect.
     // TODO: Figure out how to retry without microphone if possible.
     mediaStreamConstraints: {
-      audio: audioInput && hasAudio ? { deviceId: audioInput } : true,
-      video: videoInput ? { deviceId: videoInput } : true,
+      audio: audioInput && hasAudio ? { deviceId: audioInput } : false,
+      video: videoInput && hasVideo ? { deviceId: videoInput } : false,
```

Comment:
We need to keep the default at `true`, otherwise you can end up with `audio: false, video: false,`, which will throw an error as you need to request at least one of them.

Also, if you're a brand new user (or when you reset the site's permissions at `chrome://settings/content`), the browser won't ask for microphone or camera permission and right now we don't have a way to re-grant permission.

That means you can end up in a state where you will have camera permission but not microphone permission, with no way out of there.

Reply:
👍🏻  changed, but I will double-check how it behaves in a fresh browser session.

Reply:
I think we are okay here with true/true, the difference I saw was the size of the output file from the device, when `audio=true, video=true (just hidden)` on 10s recording was `1987172` bytes, while `audio=true, video=false` was `64623` bytes, but when converted to mp3 both ended up with `~82605` bytes.